server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -248,4 +252,15 @@ public void testRoundupFormatterLocale() {
         assertThat(roundupParser.getLocale(), is(locale));
         assertThat(formatter.locale(), is(locale));
     }
+
+    public void test0MillisAreFormatted() {
+        DateFormatter formatter = DateFormatter.forPattern("strict_date_time");
+
+        Clock clock = Clock.fixed(ZonedDateTime.of(2019,02,8,11,43,00,0,
+            ZoneOffset.UTC).toInstant(),ZoneOffset.UTC);
+
+        String formatted = formatter.formatMillis(clock.millis());
+
+        assertThat(formatted, is("2019-02-08T11:43:00.000Z"));
```

Comment:
nit: maybe ommit all the blank lines. Its a matter of taste but I tend to only use them in large methods to group semantically related blocks only.